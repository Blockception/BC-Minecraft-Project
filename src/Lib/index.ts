/*	Auto generated	*/

export * from "./mcattributes";
export * from "./mcdefinitions";
export * from "./mcignore";
export * from "./project";