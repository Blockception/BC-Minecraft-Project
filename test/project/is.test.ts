import assert = require("assert");
import { MCProject } from "../../src/main";
import { TestFilesFolder } from "../utillity.test";
import * as path from "path";

suite("MCProject", () => {
  test("is test1", () => {
    let test = MCProject.createEmpty();

    if (!MCProject.is(test)) {
      assert.fail();
    }
  });

  test("is p1", () => {
    const folder = path.join(TestFilesFolder, "mcproject", "p1");

    let project = MCProject.loadSync(folder);

    if (!MCProject.is(project)) {
      assert.fail();
    }
  });

  test("is async p1", (done) => {
    const folder = path.join(TestFilesFolder, "mcproject", "p1");

    MCProject.load(folder).then((project) => {
      if (!MCProject.is(project)) {
        assert.fail();
      }

      done();
    });
  });
});
