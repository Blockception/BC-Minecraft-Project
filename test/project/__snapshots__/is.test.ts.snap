// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`MCProject is async p1 1`] = `
{
  "attributes": {
    "diagnose": "true",
    "diagnose.mcfunctions": "true",
    "diagnose.objectives": "true",
    "diagnose.tags": "false",
    "world.area_used": "0 0 0 1000 256 1000",
  },
  "definitions": {
    "family": {
      "defined": [
        "npc",
        "Npc",
        "enemy",
      ],
      "excluded": [
        "nemesis",
      ],
    },
    "name": {
      "defined": [
        "Steve",
      ],
      "excluded": [
        "Creeper",
      ],
    },
    "objective": {
      "defined": [
        "counter",
      ],
      "excluded": [
        "Foo",
      ],
    },
    "tag": {
      "defined": [
        "tag_number1",
      ],
      "excluded": [
        "FoLolOl",
      ],
    },
  },
  "ignores": {
    "patterns": [
      "OutputFolder",
      "Temp",
      "Template/something",
      "!BP/**/*.json",
    ],
  },
}
`;

exports[`MCProject is it1 1`] = `
{
  "attributes": {},
  "definitions": {},
  "ignores": {
    "patterns": [],
  },
}
`;

exports[`MCProject is p1 1`] = `
{
  "attributes": {
    "diagnose": "true",
    "diagnose.mcfunctions": "true",
    "diagnose.objectives": "true",
    "diagnose.tags": "false",
    "world.area_used": "0 0 0 1000 256 1000",
  },
  "definitions": {
    "family": {
      "defined": [
        "npc",
        "Npc",
        "enemy",
      ],
      "excluded": [
        "nemesis",
      ],
    },
    "name": {
      "defined": [
        "Steve",
      ],
      "excluded": [
        "Creeper",
      ],
    },
    "objective": {
      "defined": [
        "counter",
      ],
      "excluded": [
        "Foo",
      ],
    },
    "tag": {
      "defined": [
        "tag_number1",
      ],
      "excluded": [
        "FoLolOl",
      ],
    },
  },
  "ignores": {
    "patterns": [
      "OutputFolder",
      "Temp",
      "Template/something",
      "!BP/**/*.json",
    ],
  },
}
`;
